fireant/queries/builder/dataset_blender_query_builder.py

```diff
@@ -137,30 +138,43 @@ def _map_fields(fields):
 
             yield field.for_(field_map[field_from_blender])
 
-    dataset_metrics = _map_fields(metrics)
-
-    # Important: If no metrics are needed from this data set, then do not query
-    if not dataset_metrics:
-        return None
-
+    dataset_metrics = ordered_distinct_list_by_attr(_map_fields(metrics))
     dataset_dimensions = _map_fields(dimensions)
     dataset_filters = _map_fields(filters)
     dataset_references = _map_fields(references)
 
-    return (
-        dataset.query()
-        .widget(EmptyWidget(*dataset_metrics), mutate=True)
-        .dimension(*dataset_dimensions, mutate=True)
-        .filter(*dataset_filters, mutate=True)
-        .reference(*dataset_references, mutate=True)
+    if not any([dataset_metrics, dataset_dimensions]):
+        return [None]
+
+    # filter out operations that are relevant for this dataset
```

Comment:
I guess the comment is meant to be 'filter out operations that are **not** relevant for this dataset'

Reply:
Makes sense